lib/galaxy/tools/parameters/basic.py

```diff
@@ -404,9 +394,16 @@ class TextToolParameter(SimpleTextToolParameter):
     def __init__(self, tool, input_source):
         input_source = ensure_input_source(input_source)
         super().__init__(tool, input_source)
+        self.profile = tool.profile if tool else None
         self.datalist = []
         for title, value, _ in input_source.parse_static_options():
             self.datalist.append({"label": title, "value": value})
+
+        # why does Integer and Float subclass this :_(
```

Comment:
why does it? :)

Reply:
I don't know but the code sharing in this file is very problematic all over.